File: 1803-average-waiting-time/average-waiting-time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        int numberOfCustomers = customers.length;
        long totalWaitingTime = 0;
        long currentTime = 0;
        for (int[] customer : customers) {
            currentTime = Math.max(currentTime, customer[0]);
            currentTime += customer[1];
            totalWaitingTime += currentTime - customer[0];
        }
        return (double) totalWaitingTime / numberOfCustomers;
    }
}

File: 1803-average-waiting-time/average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        long long current_time = 0;
        long long total_waiting_time = 0;
        for (const auto& customer : customers) {
            int arrival = customer[0];
            int time_needed = customer[1];
            if (current_time < arrival) {
                current_time = arrival;
            }
            current_time += time_needed;
            total_waiting_time += current_time - arrival;
        }
        return static_cast<double>(total_waiting_time) / customers.size();
    }
};